### PR TITLE
Udpate H2 and C6 clocks. Remove i2c_clock for all chips but ESP32

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Update clocks and add missing clocks for H2 and C6
- Align i2c0 C6 fields with H2
- Remove i2c_clocks for all chips but esp32

Gathered some information about I2C clocks in the different chips in this HackMD: https://hackmd.io/@SergioGasquez/r1dvX1AHh

Checked the I2C freq for some devices: 
- ESP32:
I2C_ESP32
- ESP32-C6:
I2C_C6
- ESP32-H2:
I2C_H2

~~Draft PR until PACs are updated (https://github.com/esp-rs/esp-pacs/pull/130 and https://github.com/esp-rs/esp-pacs/pull/129)~~

### CODE REVIEW

File: esp-hal-common/src/clock/clocks_ll/esp32h2.rs [hunk 1]
```diff
@@ -175,8 +168,7 @@ pub(crate) fn esp32h2_rtc_update_to_xtal(freq: XtalClock, _div: u8) {
         ets_update_cpu_frequency(freq.mhz());
         // Set divider from XTAL to APB clock. Need to set divider to 1 (reg. value 0)
         // first.
-        pcr.ahb_freq_conf
-            .modify(|_, w| w.ahb_div_num().bits(_div - 1));
```
Comment (SergioGasquez) [inline esp-hal-common/src/clock/clocks_ll/esp32h2.rs:179]: We were duplicating the `clk_ll_ahb_set_divider` fn

File: esp-hal-common/src/clock/mod.rs [hunk 2]
```diff
@@ -89,23 +93,44 @@ impl Clock for XtalClock {
...
+    #[cfg(esp32h2)]
+    Pll96MHz,
+    #[cfg(esp32c6)]
+    Pll160MHz,
+    #[cfg(esp32c6)]
+    Pll240MHz,
+    #[cfg(not(any(esp32c2, esp32c6, esp32h2)))]
     Pll320MHz,
+    #[cfg(not(esp32h2))]
     Pll480MHz,
```
Comment (SergioGasquez) [inline esp-hal-common/src/clock/mod.rs]: - [H2 PLL clocks](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32h2/include/hal/clk_tree_ll.h#L30-L33)
- [C6 PLL clocks](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32c6/include/hal/clk_tree_ll.h#L29-L32)

**Review by bjoernQ — APPROVED** [review]
LGTM

**Review by jessebraham — APPROVED** [review]
LGTM, thanks for cleaning this all up!

**Pull request merged**